geode-core/src/main/java/org/apache/geode/cache/query/internal/QueryConfigurationServiceImpl.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.cache.query.internal;
+
+import static org.apache.geode.distributed.internal.DistributionConfig.GEMFIRE_PREFIX;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Set;
+
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.annotations.Immutable;
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.query.internal.xml.QueryMethodAuthorizerCreation;
+import org.apache.geode.cache.query.security.JavaBeanAccessorMethodAuthorizer;
+import org.apache.geode.cache.query.security.MethodInvocationAuthorizer;
+import org.apache.geode.cache.query.security.RegExMethodAuthorizer;
+import org.apache.geode.cache.query.security.RestrictedMethodAuthorizer;
+import org.apache.geode.cache.query.security.UnrestrictedMethodAuthorizer;
+import org.apache.geode.internal.ClassPathLoader;
+import org.apache.geode.internal.cache.CacheService;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.logging.internal.log4j.api.LogService;
+import org.apache.geode.management.internal.beans.CacheServiceMBeanBase;
+
+public class QueryConfigurationServiceImpl implements QueryConfigurationService {
+
+  private static final Logger logger = LogService.getLogger();
+  static final String UPDATE_ERROR_MESSAGE =
```

Comment:
This string will need to be visible for verification of the output of gfsh commands (the next story that is being worked on) so we'll keep it package-private here to avoid having to just switch it back immediately in the next PR.

Reply:
Good to know, thanks for the explanation! 👍